debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlHistoryRecordComparator.java

```diff
@@ -143,4 +141,47 @@ else if (recordedGtidSetStr != null) {
         // The binlog coordinates are the same ...
         return true;
     }
+
+    private static class BinlogFilename implements Comparable<BinlogFilename> {
+        final private String baseName;
+        final private long extension;
+
+        private BinlogFilename(String baseName, long extension) {
+            this.baseName = baseName;
+            this.extension = extension;
+        }
+
+        public static BinlogFilename of(String filename) {
+            int index = filename.lastIndexOf(".");
+            if (index == -1) {
+                throw new IllegalArgumentException("Filename does not have an extension: " + filename);
+            }
+
+            String baseFilename = filename.substring(0, index);
+            String stringExtension = filename.substring(index + 1);
+
+            long extension;
+            try {
+                extension = Long.parseLong(stringExtension);
+            }
+            catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Can't parse binlog filename extension: " + filename, e);
+            }
+
+            return new BinlogFilename(baseFilename, extension);
+        }
+
+        @Override
+        public String toString() {
+            return "BinlogFilename [baseName=" + baseName + ", extension=" + extension + "]";
+        }
+
+        @Override
+        public int compareTo(BinlogFilename other) {
+            if (!baseName.equals(other.baseName)) {
+                throw new IllegalArgumentException("Can't compare binlog filenames with different base names");
```

Comment:
From the documentation:

> If no **base_name** value is given, the default name is the value of the `--pid-file` option (which by default is the name of host machine) followed by `-bin`.

So technically it is possible that if the server wasn't configured with an explicit base name and got its host renamed, the new binlog files will have a different name.

Although, the documentation also says:

> It is recommended that you specify a base name explicitly rather than using the default of the host name.

So it should be quite a rare case if existent at all.

Reply:
@morozov Unfortunately we cannot exlude this possibility. So I don't havy any problem toto it on the `main` branch I am not eager to push it to `1.9`

Reply:
This is not a problem. We will run a patched version until this change is released.

Reply:
@morozov Cool, thanks for that!